doc/musig-spec.mediawiki

```diff
@@ -1,38 +1,61 @@
+* '''Third party nonce aggregation''': Instead of every signer sending their nonce to every other signer, it is possible to use an untrusted third party that collects all signers' nonces, computes an aggregate nonce, and broadcasts it to the signers. This reduces the communication complexity from quadratic to linear in the number of signers. If the aggregator sends an incorrect aggregate nonce, the signing session will fail to produce a valid Schnorr signature. However, the aggregator cannot negatively affect the security of the scheme.
+* '''Partial signature verification''': If any signer sends a partial signature contribution that was not created by honestly following the protocol, the signing session will fail to produce a valid Schnorr signature. This standard specifies a partial signature verification algorithm to identify disruptive signers. It is incompatible with third-party nonce aggregation because it would be impossible to tell if a signer or the aggregator is to blame.
+* '''MuSig2* optimization''': The specification uses an optimization that allows saving a point multiplication in key aggregation. The MuSig2 scheme with this optimization is called MuSig2* and proven secure in the appendix of the [https://eprint.iacr.org/2020/1261 MuSig2 paper]. The optimization is that the second key in the list of public keys given to the key aggregation algorithm (as well as any keys identical to this key) gets the constant key aggregation coefficient ''1''.
+* '''Parameterization of MuSig2 and security''': In this specification, each signer's nonce consists of two elliptic curve points. The [https://eprint.iacr.org/2020/1261 MuSig2 paper] gives distinct security proofs depending on the number of points that constitute a nonce. The proof uses the random oracle model (ROM) for four or more points. For two points, the proof requires a stronger model, namely the combination of the ROM and the algebraic group model (AGM). As of this writing, there is no instance of a serious protocol with a security proof in the AGM that is not secure in practice. There are, however, insecure toy schemes with AGM security proofs, but those explicitly violate the requirements of the AGM<ref>The (known) [https://eprint.iacr.org/2022/226.pdf broken AGM proofs of toy schemes] feed group elements to the adversary without declaring them as group element inputs. In contrast, in MuSig2, all group elements that arise in the protocol are known to the adversary and declared as group element inputs.</ref>. A scheme very similar to MuSig2 and with two point nonces was independently proven secure in the ROM and AGM by [https://eprint.iacr.org/2020/1245 Alper and Burdges]. Consequently, it is reasonable to prefer the variant with two points.
```

Comment:
This is an important design decision that needs a justification and it should be either discussed in or linked from this BIP. Even if we didn't discuss it, this will come up at some point and I prefer having a linkable writeup somewhere over a bunch of twitter answers.

Reply:
I moved the argument to a new section in the remarks and added one of your arguments.